pkgs/tools/typesetting/tex/texlive/combine.nix

```diff
@@ -311,5 +336,3 @@ in (buildEnv {
   ''
   ;
 }).overrideAttrs (_: { allowSubstitutes = true; })
-# TODO: make TeX fonts visible by fontconfig: it should be enough to install an appropriate file
-#       similarly, deal with xe(la)tex font visibility?
```

Comment:
the mere existence of this fontconfig file does not really change anything, xelatex still won't find texlive fonts:
```
➜ cat b.tex 
\documentclass{article}
\usepackage{mwe,fontspec}
\setmainfont{STIX Two Text}
\begin{document}
\Blinddocument
\end{document}

ulli on pumpkin nixpkgs (b737e7f) [$] 
➜ ./result/bin/xelatex b.tex
This is XeTeX, Version 3.141592653-2.6-0.999994 (Web2C 2022/nixos.org) (preloaded format=xelatex)
 restricted \write18 enabled.
entering extended mode
(./b.tex
[...]
(/nix/store/g3r3yhkbaar2syfjdhlbhair1nw43kkw-texlive-combined-full-2022.2022122
7-texmfdist/tex/latex/fontspec/fontspec.cfg)))

! Package fontspec Error: The font "STIX Two Text" cannot be found.

For immediate help type H <return>.
 ...                                              
                                                  
l.4 \begin
          {document}
? X
No pages of output.
Transcript written on b.log.
```

To fix this, this fontconfig file would have to be included in the system's fontconfig (which obviously cannot be done from the derivation, but we could provide a nixos module that does this or just adds $TEXMFDIST/fonts to `fonts.fonts`) or passed in the `FONTCONFIG_FILE` environment variable:
```
✗ FONTCONFIG_FILE=$(realpath result/share/texmf-var/fonts/conf/texlive-fontconfig.conf) xelatex b.tex
This is XeTeX, Version 3.141592653-2.6-0.999994 (Web2C 2022/nixos.org) (preloaded format=xelatex)
 restricted \write18 enabled.
entering extended mode
[...]
Output written on b.pdf (16 pages).
Transcript written on b.log.
```
But we cannot wrap xe[la]tex with this since this would prevent the usage of system fonts, i.e. completely invert the current behaviour (XeTeX finds system fonts, but no texlive fonts).

Not sure whether adding a nixos module with the sole purpose of adding something to the font search path is apporiate, but we should probably document somewhere visible how to make texlive fonts visible to fontconfig/xetex (maybe the `longDescription` field of the combined derivations, it appears rather visibly on search.nixos.org). This is not only useful for xetex users (apparently these still exist), but also for users of other typesetting software.

In any case, this is hardly related to this PR

Reply:
The TODO is back, just further up next to the postactions. There is a new comment based on details found in `texmf.cnf`: allegedly, the new `fonts.conf` is read by MkIV (but not by xetex, by design).

A solution to the xetex problem is probably to attach a `fonts` attribute to the output of `texlive.combine` (sort of a split output) which symlinks the font folder, for inclusion in the `fonts.fonts` option. I don't use NixOS so I won't attempt a PR, I wouldn't be able to test it.

Reply:
Fontconfig issue fixed in #228196